### PR TITLE
Add nats-server application build from source

### PR DESCRIPTION
**Pull request by SHOREQU69**
Addresses issue: https://github.com/unikraft/dynamic-apps/issues/31

Co-authored-by: MekalBoy <23715925+MekalBoy@users.noreply.github.com>

### CODE REVIEW

**Review by StefanJum — CHANGES_REQUESTED** [review]
@SHOREQU69 you don't need to add the binaries, just the `build.sh` script.
Run the script through [shellcheck](https://www.shellcheck.net/).
Replace `cd x/y/z` with `pushd x/y/z &> /dev/null || exit 1` and `cd ..` with `popd &> /dev/null || exit 1`

**Review by RaduNichita — COMMENTED** [review]
Nice job, @SHOREQU69. 

Please change your email addreses in the commit message and see the other comments below.

File: nats-server/build.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+cd ${name}-${ver}
+go build -ldflags "-linkmode 'external' -extldflags '-pie'" -buildmode=pie
+cd ..
+cp ${name}-${ver}/${name} .
+rm -rf ${name}-${ver}
+
+echo Extracting libs
+$extract ${name}
+
+echo "Finished"
```
[the remark this answers is not in the document]
Comment (SHOREQU69) [reply nats-server/build.sh:35]: What do you mean by changing the email addreses? Is it the users.noreply.github.com?

[the remark this answers is not in the document]
Comment (RaduNichita) [reply nats-server/build.sh:35]: Yeah, it should be your real email address.

**Review by StefanJum — APPROVED** [review]
All good, thanks @SHOREQU69.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**